clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -5317,6 +5318,12 @@ bool TokenAnnotator::mustBreakBefore(const AnnotatedLine &Line,
   if (Right.NewlinesBefore > 1 && Style.MaxEmptyLinesToKeep > 0)
     return true;
 
+  if (Style.BreakFunctionDefinitionParameters && Line.MightBeFunctionDecl &&
```

Comment:
Are all those checks needed?

Reply:
Removed one unneeded check, the rest are all needed.